association/association_analysis.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import pandas as pd
+from mlxtend.frequent_patterns import apriori
+from mlxtend.frequent_patterns import association_rules
+
+
+def create_tx_map(df):
+    # (k, v): (transaction_id, list(index))
+    tx_map = {}
+    for i, cols in df.iterrows():
+        key = df['transactionId'][i]
+        tx_map.setdefault(key, [])
+        tx_map[key].append(i)
+    return tx_map
+
+
+def is_positif(x):
+    return 1 if x > 0 else 0
+
+
+def multi_dept_tx(dic, df1, c1, c2, c3, c4):
+    for k, v in dic.items():
+        if len(v) != 1:
+            for x in v:
+                c1.append(df1.iloc[x, 0])
+                c2.append(df1.iloc[x, 1])
+                c3.append(df1.iloc[x, 2])
+                c4.append(df1.iloc[x, 3])
```

Comment:
It's better to drop rows using index `i`, when the dictionary contains only one element:

```py
if len(indexes) == 1:
    i = indexes[0]
    df.drop(df.index[i])
```